packages/mapviewer/src/modules/infobox/components/FeatureDetail.vue

```diff
@@ -1,59 +1,87 @@
 
-const { feature } = defineProps({
+const props = defineProps({
     feature: {
         type: SelectableFeature,
         required: true,
     },
 })
 
+const { feature } = toRefs(props)
+
 const { t } = useI18n()
 
 const store = useStore()
-const hasFeatureStringData = computed(() => typeof feature?.data === 'string')
-const popupDataCanBeTrusted = computed(() => feature.popupDataCanBeTrusted)
+const hasFeatureStringData = computed(() => typeof feature.value?.data === 'string')
+const popupDataCanBeTrusted = computed(() => feature.value.popupDataCanBeTrusted)
 
 const coordinateFormat = computed(() => {
     return allFormats.find((format) => format.id === store.state.position.displayedFormatId) ?? null
 })
 const sanitizedFeatureDataEntries = computed(() => {
-    if (hasFeatureStringData.value || !feature?.data) {
+    if (hasFeatureStringData.value || !feature.value?.data) {
         return []
     }
-    return Object.entries(feature.data)
-        .filter(([_, value]) => value) // filtering out null values
+    return Object.entries(feature.value.data)
+        .filter(([_, value]) => value) // Filtering out null values
         .map(([key, value]) => [
             key,
             sanitizeHtml(value, key === 'description'),
-            getIframeHosts(value),
+            key === 'description' ? getIframeHosts(value) : [],
         ])
 })
 function sanitizeHtml(htmlText, withIframe = false) {
+    const blockedExternalContentString = t('blocked_external_content')
     DOMPurify.addHook('afterSanitizeAttributes', function (node) {
-        // set all elements owning target to target=_blank
-        if ('target' in node) {
+        // Replacing possibly malicious code instead of removing with node.remove()
+        if (node.tagName === 'A') {
             node.setAttribute('target', '_blank')
-            node.setAttribute('rel', 'noopener')
+            node.setAttribute('rel', 'noopener noreferrer')
+            try {
+                const url = new URL(node.getAttribute('href'))
+                const ext = url.pathname.split('.').pop().toLowerCase()
+                if (BLOCKED_EXTENSIONS.includes(ext)) {
+                    node.outerHTML = blockedExternalContentString
+                    return
+                }
+            } catch (error) {
+                log.error(error)
+                node.outerHTML = blockedExternalContentString
+                return
+            }
+        }
+        if (node.tagName === 'IFRAME') {
+            try {
+                const src = new URL(node.getAttribute('src'))
+                const ext = src.pathname.split('.').pop().toLowerCase()
+                if (BLOCKED_EXTENSIONS.includes(ext)) {
+                    node.outerHTML = blockedExternalContentString
+                    return
+                }
+            } catch (error) {
+                log.error(error)
+                node.outerHTML = blockedExternalContentString
+                return
+            }
```

Comment:
the only thing you do differently in these two cases is how you select the "content" (once `const url = new URL(node.getAttribute('href'))` and the other is `const src = new URL(node.getAttribute('src'))`)
Can you find a way to have the next part of the code only written once?

And maybe explain a little bit the security concern you are dealing here in the code with a little comment before all that.